### PR TITLE
[project-base] friendly url fixes

### LINKED ISSUE #3107: Unused friendly urls for flags returns 500
## Describe the bug

If there is an unused friendly url for a flag, you'll get an error 500 after opening the link instead of expected 404 page or redirect.

Another problem is, when you choose a permanent or temporary redirect for any of the unused friendly urls, you aren't redirect anywhere. Instead of it you'll get the 404 page.

Then If you want to create a flag with your previously deleted unused friendly url, you'll get an error 500 in the process

### PR DESCRIPTION
**Pull request by vitek-rostislav**
| Q             | A
| ------------- | ---
|Description, reason for the PR| There are several bugs in the work with friendly URLs - the redirect does not work, and 500 error occur in various scenarios, see #3107 
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues|  closes #3107 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://rv-friendly-url.odin.shopsys.cloud
  - https://cz.rv-friendly-url.odin.shopsys.cloud

### CODE REVIEW

File: project-base/app/src/Component/Router/FriendlyUrl/FriendlyUrlMatcher.php [hunk 1]
```diff
@@ -49,6 +49,7 @@ public function match($pathinfo, RouteCollection $routeCollection, DomainConfig
             $matchedParameters['_controller'] = 'Symfony\Bundle\FrameworkBundle\Controller\RedirectController::urlRedirectAction';
```
Comment (grossmannmartin) [inline project-base/app/src/Component/Router/FriendlyUrl/FriendlyUrlMatcher.php:51]: the $matchedParameters array is created dynamically if I'm not mistaken... Is it ok?

File: project-base/app/src/Model/Product/Flag/FlagRepository.php [hunk 2]
```diff
@@ -63,7 +63,13 @@ public function getVisibleFlagById(int $flagId, string $locale): Flag
             ->setParameter('flagId', $flagId)
             ->setParameter('locale', $locale);
 
-        return $flagsQueryBuilder->getQuery()->getSingleResult();
+        $flag = $flagsQueryBuilder->getQuery()->getOneOrNullResult();
+
+        if ($flag === null) {
+            throw new FlagNotFoundException(sprintf('Flag with ID "%s" does not exist.', $flagId));
```
Comment (grossmannmartin) [inline project-base/app/src/Model/Product/Flag/FlagRepository.php:69]: can it still be a problem in flagBreadcrumbGenerator? The exception is not caught there...

**Review by grossmannmartin — CHANGES_REQUESTED** [review]

File: project-base/app/src/Component/Router/FriendlyUrl/FriendlyUrlMatcher.php (hunk 1 above)
Comment (vitek-rostislav) [reply project-base/app/src/Component/Router/FriendlyUrl/FriendlyUrlMatcher.php:51]: good catch, fixed in 25eea475d1

File: project-base/app/src/Model/Product/Flag/FlagRepository.php (hunk 2 above)
Comment (vitek-rostislav) [reply project-base/app/src/Model/Product/Flag/FlagRepository.php:69]: that is a good point, added c190860950 commit

**Review by grossmannmartin — APPROVED** [review]

**Review by pk16011990 — APPROVED** [review]

**Pull request merged**